x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/use_confusion_matrix.ts

```diff
@@ -25,13 +27,45 @@ import {
 
 import { isTrainingFilter } from './is_training_filter';
 
+function getEvalutionMetricsItems(evalMetrics?: ClassificationEvaluateResponse['classification']) {
+  if (evalMetrics === undefined) return [];
+
+  const accuracyMetrics = evalMetrics.accuracy?.classes || [];
+  const recallMetrics = evalMetrics.recall?.classes || [];
+
+  const metricsMap = accuracyMetrics.reduce((acc, accuracyMetric) => {
+    acc[accuracyMetric.class_name] = {
+      className: accuracyMetric.class_name,
+      accuracy: accuracyMetric.value,
+    };
+    return acc;
+  }, {} as Record<string, ClassificationMetricItem>);
+
+  recallMetrics.forEach((recallMetric) => {
+    if (metricsMap[recallMetric.class_name] !== undefined) {
+      metricsMap[recallMetric.class_name] = {
+        recall: recallMetric.value,
+        ...metricsMap[recallMetric.class_name],
+      };
+    } else {
+      metricsMap[recallMetric.class_name] = {
+        className: recallMetric.class_name,
+        recall: recallMetric.value,
+      };
+    }
+  });
+
+  return Object.values(metricsMap);
+}
+
 export const useConfusionMatrix = (
   jobConfig: DataFrameAnalyticsConfig,
   searchQuery: ResultsSearchQuery
 ) => {
   const [confusionMatrixData, setConfusionMatrixData] = useState<ConfusionMatrix[]>([]);
   const [overallAccuracy, setOverallAccuracy] = useState<null | number>(null);
   const [avgRecall, setAvgRecall] = useState<null | number>(null);
+  const [evaluationMetricsItems, setEvalutaionMetricsItems] = useState<any>([]);
```

Comment:
Please avoid using `any`

Reply:
Thanks for taking a look 🙏  Good catch - updated in eb6c96330a7daa839e9cb879010edf6f53572f0c